packages/core/services/AnnotationService/DatabaseAnnotationService/index.ts

```diff
@@ -81,23 +76,35 @@ export default class DatabaseAnnotationService implements AnnotationService {
                 }
             });
 
+        return this.fetchFilteredValuesForAnnotation(hierarchy[path.length], filtersByAnnotation);
+    }
+
+    private async fetchFilteredValuesForAnnotation(
+        annotation: string,
+        filtersByAnnotation: { [name: string]: (string | null)[] } = {}
+    ): Promise<string[]> {
         const sqlBuilder = new SQLBuilder()
-            .select(`DISTINCT "${hierarchy[path.length]}"`)
+            .select(`DISTINCT "${annotation}"`)
             .from(this.dataSourceNames);
 
-        Object.keys(filtersByAnnotation).forEach((annotation) => {
-            const annotationValues = filtersByAnnotation[annotation];
+        Object.keys(filtersByAnnotation).forEach((annotationToFilter) => {
+            const annotationValues = filtersByAnnotation[annotationToFilter];
             if (annotationValues[0] === null) {
-                sqlBuilder.where(`"${annotation}" IS NOT NULL`);
+                sqlBuilder.where(`"${annotationToFilter}" IS NOT NULL`);
             } else {
                 sqlBuilder.where(
-                    annotationValues.map((value) => `"${annotation}" = '${value}'`).join(") OR (")
+                    annotationValues
+                        .map((value) => `"${annotationToFilter}" = '${value}'`)
```

Comment:
Not sure if I'm commenting in the right place, but noticed that when doing a grouped query (e.g., group by color), if a file has multiple values (red, green, blue) and you filter on just color=green it won't return that file. When no grouping is applied color=green returns the red,green,blue file

Reply:
Resolved in https://github.com/AllenInstitute/aics-fms-file-explorer-app/pull/186/commits/4df670694af1d9232c63892b2ea2d2dc0b8172a1
I wasn't using regex when grabbing annotation values to use as folders in the heirarchy